satellite/marketingweb/server.go

```diff
@@ -180,6 +184,24 @@ func (s *Server) CreateOffer(w http.ResponseWriter, req *http.Request) {
 	http.Redirect(w, req, "/", http.StatusSeeOther)
 }
 
+// StopOffer expires the current offer and replaces it with the default offer.
+func (s *Server) StopOffer(w http.ResponseWriter, req *http.Request) {
+	offerID, err := strconv.Atoi(mux.Vars(req)["offer_id"])
+	if err != nil {
+		s.log.Error("failed to parse offer id", zap.Error(err))
+		s.serveBadRequest(w, req, err)
+		return
+	}
+
+	if err := s.db.Finish(req.Context(), offerID); err != nil {
```

Comment:
The declaration of `Finish()` has changed to `Finish(context, offerID, isDefault)`